engine/dscache.go

```diff
@@ -110,19 +112,11 @@ func (c *dsCache) Sync(key ds.Key) error {
 	return c.dstore.Sync(key)
 }
 
-// Close implements datastore interface.
+// Close always returns nil.
+// Note that calls to this function does not close the underlying datastore.
+// This is because dsCache wraps an existing datastore and does not construct it, and the wrapped
+// datastore may be in use elsewhere.
 func (c *dsCache) Close() error {
-	c.lock.Lock()
-	defer c.lock.Unlock()
-
-	if c.dstore == nil {
-		return nil
-	}
-
-	if err := c.dstore.Close(); err != nil {
-		return err
-	}
-	c.dstore = nil
 	return nil
```

Comment:
I think that it is at least necessary to do a `c.dstore.Sync()` here as it may be expected that things are persisted to disk after close.

Reply:
I agree.

Reply:
Good call thank you